api/provider/aws/environment_delete.go

```diff
@@ -53,6 +59,61 @@ func (e *EnvironmentProvider) Delete(environmentID string) error {
 	return nil
 }
 
+func (e *EnvironmentProvider) checkEnvironmentDependencies(environmentID string) error {
+	loadBalancerTags, err := e.TagStore.SelectByType("load_balancer")
+	if err != nil {
+		return err
+	}
+
+	dependentLoadBalancerTags := loadBalancerTags.WithKey("environment_id").WithValue(environmentID)
```

Comment:
all of this logic follows the same pattern: get a slice of tags, filter by `WithKey("environment_id").WithValue(environmentID)`, if the length of the filtered tags are > 0, generate a new slice from the filtered tags, then throw an error

Reply:
I think we can reduce a lot of this code given that it all follows a similar pattern.

Reply:
✅